src/layouts/BasicLayout.js

```diff
@@ -64,6 +64,27 @@ class BasicLayout extends React.PureComponent {
     // find the current route from menuData.
     const currRouteMap = this.breadcrumbNameMap.find(route => route.path.endsWith(pathname));
 
+    // eslint-disable-next-line no-undef
+    fetch(ENDPOINT)
```

Comment:
I still don't understand what you're doing with `ENDPOINT`. Your config change has it pointing at `localhost:8000` which I wouldn't expect will ever respond. For real production deployments we want to fetch from `/api/v1/endpoints` (implicitly `window.origin`) while for your local development mode you want to point to the real endpoint of an actual Pbench server, like `http://10.1.170.1/api/v1/endpoints`. You're not doing either. Are you intending to build an npm "reflector" that does handle `http://localhost:8000/api/v1/endpoints` by calling a real `http://10.1.170.1/api/v1/endpoints` (as we'd discussed once before) and returning the result? And, if so, where is that code and where is the actual remote address defined?

The code as presented in this PR will break both modes, so I'm still confused about where you plan to take this.

Reply:
Yeah, in the language of umijs, the variables that you **define** in `config/config.js` are publicly available in every js file. So this `ENDPOINT` variable is defined in the config file. When you wish to connect to the real server, you replace this `ENDPOINT` variable in the `config/config.js` to "http://10.1.170.1".

Reply:
We cannot use `window` object in a nodejs build such as `config/config.js`. It will always be undefined and wont start the yarn web server.

Reply:
Sure; but `ENDPOINT = /api/v1/endpoints` will work just fine. It's only for local developer mode that there's a need to override the default `window.origin` host/port.